cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -293,14 +294,33 @@ func (g *gitHubRepository) getVersions() ([]string, error) {
 			}
 			return false, nil
 		}
+		allReleases = append(allReleases, releases...)
+
+		// https://github.com/google/go-github/blob/master/github/github.go#L536-L546
+		if response.NextPage != 0 && response.LastPage != 0 {
+			nextPage := response.NextPage
+			lastPage := response.LastPage
+			for pageNum := nextPage; pageNum <= lastPage; pageNum++ {
+				releases, _, listReleasesErr = client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, &github.ListOptions{Page: pageNum})
+				if listReleasesErr != nil {
+					retryError = g.handleGithubErr(listReleasesErr, "failed to get the list of releases")
+					// return immediately if we are rate limited
+					if _, ok := listReleasesErr.(*github.RateLimitError); ok {
+						return false, retryError
+					}
+					return false, nil
+				}
+				allReleases = append(allReleases, releases...)
```

Comment:
This lines seem to be almost the same as 208 to 297.

Could we maybe combine these in a single for loop?

Reply:
I had them duplicated to apply the rate limiting to each call to the ListReleases API.